server/chat/backend/agent/tools/iac/iac_commands_tool.py

```
@@ -43,6 +44,58 @@
+    Returns list of (command, verdict) tuples for denied commands only.
+    Best-effort regex extraction; primary protection is the tool-level gates.
+    """
+    if not org_id:
+        return []
+
+    from utils.auth.command_policy import evaluate_compound_command
+    from utils.terminal.terminal_run import terminal_run
+
+    safe_dir = shlex.quote(terraform_dir)
+    list_cmd = f"cat {safe_dir}/*.tf 2>/dev/null || true"
+    result = terminal_run(list_cmd, shell=True, capture_output=True, text=True, timeout=15)
```

Ruff (0.15.10)

[error] Function call with `shell=True` parameter identified, security issue

(S604)